core/src/main/java/tech/tablesaw/conversion/smile/SmileConverter.java

```diff
@@ -22,125 +24,62 @@ public SmileConverter(Relation table) {
-    return dataset(
-        table.numberColumn(responseColName),
-        AttributeType.NOMINAL,
-        table.numericColumns().stream()
-            .filter(c -> !c.name().equals(responseColName))
-            .collect(Collectors.toList()));
-  }
-
-  /**
-   * Returns a dataset where the response column is nominal. E.g. to be used for a classification
-   */
-  public AttributeDataset nominalDataset(int responseColIndex, int... variablesColIndices) {
-    return dataset(
-        table.numberColumn(responseColIndex),
-        AttributeType.NOMINAL,
-        table.columns(variablesColIndices));
-  }
-
-  /**
-   * Returns a dataset where the response column is nominal. E.g. to be used for a classification
-   */
-  public AttributeDataset nominalDataset(String responseColName, String... variablesColNames) {
-    return dataset(
-        table.numberColumn(responseColName),
-        AttributeType.NOMINAL,
-        table.columns(variablesColNames));
-  }
-
-  private AttributeDataset dataset(
-      NumericColumn<?> responseCol, AttributeType type, List<Column<?>> variableCols) {
-    List<Column<?>> convertedVariableCols =
-        variableCols.stream()
-            .map(col -> col.type() == ColumnType.STRING ? col : table.nCol(col.name()))
-            .collect(Collectors.toList());
-    Attribute responseAttribute =
-        type == AttributeType.NOMINAL
-            ? colAsNominalAttribute(responseCol)
-            : new NumericAttribute(responseCol.name());
-    AttributeDataset dataset =
-        new AttributeDataset(
-            table.name(),
-            convertedVariableCols.stream()
-                .map(col -> colAsAttribute(col))
-                .toArray(Attribute[]::new),
-            responseAttribute);
-    for (int i = 0; i < responseCol.size(); i++) {
-      final int r = i;
-      double[] x =
-          IntStream.range(0, convertedVariableCols.size())
-              .mapToDouble(c -> getDouble(convertedVariableCols.get(c), dataset.attributes()[c], r))
-              .toArray();
-      dataset.add(x, responseCol.getDouble(r));
-    }
-    return dataset;
-  }
-
-  private double getDouble(Column<?> col, Attribute attr, int r) {
-    if (col.type() == ColumnType.STRING) {
-      String value = ((StringColumn) col).get(r);
-      try {
-        return attr.valueOf(value);
-      } catch (ParseException e) {
-        throw new IllegalArgumentException("Error converting " + value + " to nominal", e);
+  public DataFrame toDataFrame(StructType schema) {
+    List<Tuple> rows = new ArrayList<>();
+    int colCount = table.columnCount();
+    for (int rowIndex = 0; rowIndex < table.rowCount(); rowIndex++) {
+      Object[] row = new Object[colCount];
+      for (int colIndex = 0; colIndex < colCount; colIndex++) {
+        Column<?> col = table.column(colIndex);
+        if (!col.isMissing(rowIndex)) {
+          row[colIndex] =
+              col.type().equals(ColumnType.INSTANT)
```

Comment:
@haifengl this was the only thing I wasn't sure about in creating Tablesaw's Smile integration. There's no `InstantDataType` in Smile, so I converted the `Instant` to `LocalDateTime`. Does that seem like the right approach to you? Or would it be better for Smile to have an `InstantDataType`?

Reply:
You can wrap anything into a `DataFrame`'s Object column data type. For most use case, it should be fine to convert it to LocalDataTime too.